### PR TITLE
Feature: add optional layout

### PR DESCRIPTION
**Pull request by akushniruk**
not merge, while we have unsolved issues.
**build size was increase on 5.78MiB: style.9f6eb86ca7835016bb2d.css (5.78 MiB)**

**TODO:**

- [x] add logic for logout
- [x] add purge for tailwindcss (need investigate why it's dropped all styles during the build)
- [x] docs

### CODE REVIEW

**Comment by akushniruk** [comment]
not merge, while we have unsolved issues.
build size increase on5,78MiB: style.9f6eb86ca7835016bb2d.css (5.78 MiB)

File: handlers/index.go [hunk 1]
```diff
@@ -144,10 +145,10 @@ func index(ctx *gin.Context) {
...
 
 	ctx.HTML(http.StatusOK, "index", gin.H{
-		"title":    	"BaseApp",
-		"cssFiles": 	cssFiles,
-		"jsFiles":  	jsFiles,
-		"rootID":   	"root",
+		"title":        "BaseApp",
+		"cssFiles":     cssFiles,
+		"jsFiles":      jsFiles,
+		"rootID":       "root",
```
Comment (akhlopiachyi) [inline handlers/index.go:151]: revert

File: web/postcss.config.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const tailwindcss = require('tailwindcss');
+
+module.exports = {
+    plugins: [
+      tailwindcss,
+    ]
+  }
+
```
Comment (akhlopiachyi) [inline web/postcss.config.js:8]: End of this file is strange
Please fix it

File: web/src/api/config.ts [hunk 3]
```diff
@@ -41,10 +41,11 @@ export const defaultConfig: Config = {
     ],
     captcha_type: 'none',
     password_min_entropy: 0,
-    wizard_step: undefined,
+    wizard_step: "false",
```
Comment (akhlopiachyi) [inline web/src/api/config.ts:44]: use singlequotes

File: web/src/api/config.ts [hunk 4]
```diff
@@ -41,10 +41,11 @@ export const defaultConfig: Config = {
...
     barong_upload_size_min_range: '1',
     barong_upload_size_max_range: '20',
     themeSwitcher: 'visible',
+    useSharedLayout: "false",
```
Comment (akhlopiachyi) [inline web/src/api/config.ts:48]: use singlequotes

File: web/src/containers/Header/index.tsx [hunk 5]
```diff
@@ -68,24 +68,28 @@ class Head extends React.Component<Props> {
...
+                                onClick={this.openSidebar}>
+                                <span className="pg-sidebar__toggler-item" />
+                                <span className="pg-sidebar__toggler-item" />
+                                <span className="pg-sidebar__toggler-item" />
+                            </div>
+                            <div onClick={(e) => this.redirectToLanding()} className="pg-header__logo">
+                                <Logo />
+                            </div>
+                        </>
+                    }
```
Comment (akhlopiachyi) [inline web/src/containers/Header/index.tsx:84]: maybe let's rewrite this component using functional approach and memorize this block using useMemo

File: web/src/helpers/stringToHTML.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import dompurify from 'dompurify';
+
+export const stringToHTML = (str: string) => {
+	const sanitizer = dompurify.sanitize;
+	
+	return <div dangerouslySetInnerHTML={{__html: sanitizer(str)}}></div>;
```
Comment (akhlopiachyi) [inline web/src/helpers/stringToHTML.tsx:7]: https://www.npmjs.com/package/html-react-parser

Please, check this
Maybe better we can use this library for transforming string to html

File: web/tailwind.config.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                "color-contrast": "#FFFFFF",
+                "cta-contrast": "#090909",
+                "secondary-color": "#737373",
+            },
+            boxShadow: {
+                "lg-updated": " 0px 0px 2px rgba(0, 0, 0, 0.12), 0px 10px 15px -3px rgba(0, 0, 0, 0.12), 0px 4px 6px -2px rgba(0, 0, 0, 0.04)",
+            },
+            borderColor: {
+                "divider-color-20": "1px solid #DCDCDC",
+            },
```
Comment (akhlopiachyi) [inline web/tailwind.config.js:24]: maybe we can split this files and create a separate file: themeSettings.js, where we move all these objects: colors, textColor, boxShadow and borderColor

**Review by Kohelbekker — APPROVED** [review]

File: web/src/components/SharedLayout/index.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+const FooterProps = {
+    options: FOOTER_OPTIONS,
+    className: "bg-main-background-color",
+    dropdownBlockClassName: "absolute z-50 bottom-8 w-40 bg-white rounded shadow r-2 select-none",
+    dropdownOptionClassName: "block text-gray-400 hover:text-gray-500 hover:bg-gray-50 rounded px-4 py-2",
+    optionClassName: "text-gray-400 hover:text-gray-500 px-5 relative",
+    socialOptionClassName: "text-gray-400 hover:text-gray-500 px-2 opacity-60 hover:opacity-100 delay-200",
+    socialMoreButtonClassName: "self-center ml-2 relative bg-gray-100 rounded flex py-1 px-2 cursor-pointer",
+    socialMoreTextClassName: "self-center text-gray-500 text-sm font-bold",
+    socialMoreClassName: "absolute flex flex-wrap flex-row flex-row bottom-8 right-0 w-52 z-50 p-4 bg-white rounded shadow r-2",
```
Comment (akhlopiachyi) [inline web/src/components/SharedLayout/index.tsx:31]: use single quotes

File: web/tailwind.config.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        "color-contrast": "#FFFFFF",
+        "cta-contrast": "#090909",
+        "secondary-color": "#737373",
+    },
+    boxShadow: {
+        "lg-updated": " 0px 0px 2px rgba(0, 0, 0, 0.12), 0px 10px 15px -3px rgba(0, 0, 0, 0.12), 0px 4px 6px -2px rgba(0, 0, 0, 0.04)",
+    },
+    borderColor: {
+        "divider-color-20": "1px solid #DCDCDC",
+    }
```
Comment (akhlopiachyi) [inline web/tailwind.config.js:18]: use single quotes

File: web/tailwind.config.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        "dropdown-background-color": "#FFFFFF",
+        "main-background-color": "#FFFFFF",
+    },
+    textColor: {
+        "color-contrast": "#FFFFFF",
+        "cta-contrast": "#090909",
+        "secondary-color": "#737373",
+    },
+    boxShadow: {
+        "lg-updated": " 0px 0px 2px rgba(0, 0, 0, 0.12), 0px 10px 15px -3px rgba(0, 0, 0, 0.12), 0px 4px 6px -2px rgba(0, 0, 0, 0.04)",
```
Comment (akhlopiachyi) [inline web/tailwind.config.js:14]: remove space before `0px`

**Review by akhlopiachyi — APPROVED** [review]

**Pull request merged**